Source/Core/Duality/Utility/ReflectionHelper.cs

```diff
@@ -121,7 +121,10 @@ private static class StaticLookup<T>
 				}
 
 				// Mind the result for later. Don't do this twice.
-				customMemberAttribCache[member] = result;
+				lock (customMemberAttribCache)
+				{
+					customMemberAttribCache[member] = result;
+				}
```

Comment:
Commenting here, but this actually counts for all similar cases in this PR as well: I believe you actually need to lock both the `TryGetValue` and the item change, since otherwise you could still change the dictionary in thread A while thread B is in the middle of `TryGetValue`.

The easiest way to go about this for now would be to just expand the `lock` block around the entire "try, else update" section. If this was highly concurrent code, it might make sense to do two separate lock statements for try and update, but I doubt it's worth the hassle in this case.

Reply:
Expanded the lock to include reads from the collections, while not including any unnecessary, thread-safe code.